src/python/WMCore/Services/DBS/DBS3Reader.py

```diff
@@ -57,12 +57,13 @@ class DBS3Reader(object):
     # cache all the datatiers known by DBS
     _datatiers = {}
 
-    def __init__(self, url, **contact):
+    def __init__(self, url, logger=None, **contact):
 
         # instantiate dbs api object
         try:
             self.dbsURL = url
             self.dbs = DbsApi(url, **contact)
+            self.logger = logger or logging
```

Comment:
Isn't it `logging.getLogger()`?

Reply:
I am  always confused how to the log should be set. getLogger create new logger instance from the logging which has original setting. I think it is better to use new instance for each module with name to find which module is logging but I don't think current code will have failure.